datamodel/src/main/kotlin/com/hartwig/actin/datamodel/molecular/PanelSpecifications.kt

```diff
@@ -1,15 +1,26 @@
 package com.hartwig.actin.datamodel.molecular
 
+import java.util.function.Predicate
+
 data class PanelGeneSpecification(val geneName: String, val targets: List<MolecularTestTarget>)
 
-class PanelSpecifications(private val panelSpecifications: Map<String, List<PanelGeneSpecification>>) {
+data class PanelSpecification(private val geneTargetMap: Map<String, List<MolecularTestTarget>>) {
+
+    fun testsGene(gene: String, molecularTestTargets: Predicate<List<MolecularTestTarget>>) =
+        geneTargetMap[gene]?.let { molecularTestTargets.test(it) } ?: false
+}
+
+class PanelSpecifications(panelGeneSpecifications: Map<String, List<PanelGeneSpecification>>) {
+
+    private val panelSpecifications: Map<String, PanelSpecification> = panelGeneSpecifications.entries.associate {
+        it.key to PanelSpecification(
+            it.value.groupBy(PanelGeneSpecification::geneName).mapValues { v -> v.value.flatMap { c -> c.targets } })
+    }
```

Comment:
Tried to clean this up a bit:
```suggestion
    private val panelSpecifications: Map<String, PanelSpecification> = panelGeneSpecifications.mapValues { (_, specs) ->
        PanelSpecification(
            specs.groupBy(PanelGeneSpecification::geneName).mapValues { it.value.flatMap(PanelGeneSpecification::targets) })
    }
```